src/components/forms/Input.tsx

```diff
@@ -88,14 +97,31 @@ export function Input(props: InputProps) {
     error,
     valid,
     loading,
+    clearable = false,
+    onChange,
     ...otherProps
   } = props;
+  const ref = useRef<HTMLInputElement>(null);
 
   const { name, variant: contextVariant } = useFieldsContext();
 
   const hasLeading = (leadingAddon && !leadingAddon.inline) || false;
   const hasTrailing = (trailingAddon && !trailingAddon.inline) || false;
   const variant = variantProps || contextVariant;
+  const clearInput = () => {
+    if (ref.current) {
+      setValue('');
+      ref.current.value = '';
+      const event = {
+        target: ref.current,
+      } as ChangeEvent<HTMLInputElement>;
+      onChange?.(event);
+    }
+  };
+  const [value, setValue] = useState(otherProps.value);
+  useEffect(() => {
+    setValue(otherProps.value);
+  }, [otherProps.value]);
```

Comment:
The component is controlled and should not have this kind of internal state or a `useEffect`

I think the problem your are running into is the `event` param of `onChange`, which does not play nice with this new `clearable` option.

I've done a little bit of research, and ant has a similar api, and they use a library which solves it like this: https://github.com/react-component/input/blob/aa08e91ce278103f3d0b5d1642ee541fe87e7ad2/src/utils/commonUtils.ts#L12

I don't like this solution. It pretends that the event originates from the input when it actually comes from the clear button.

My suggestion is to make a breaking change:
`onChange` -> `onValueChange`
`onChange` cannot be passed down from the `Input` to the `input`. It is created internally and forwards events to `onValueChange`.

```ts
interface InputProps {
  // The second argument is passed only when the change in value
  // comes from the input's onChange event
  onValueChange: (value: string, event?: React.ChangeEvent<HTMLInputElement>);
}

@targos what do you think?

Reply:
Shouldn't we just change the value and use `dispatchEvent` to dispatch the change event?

Reply:
I also read in many places that we should dispatch an `input` event: https://stackoverflow.com/questions/23892547/what-is-the-best-way-to-trigger-change-or-input-event-in-react-js

Reply:
> Shouldn't we just change the value and use `dispatchEvent` to dispatch the change event?

IDK, really not sure about what to do with this one. By doing this you can end up in a situation where the `value` prop of the `Input` and the actual value of the `input` are not synced. Is it acceptable?

Reply:
Isn't it what happens already with React when a component is controlled? When you listen to the change event, `event.target.value` is already changed.

Reply:
Yes it's true, haven't thought about it that way.

@wadjih-bencheikh18 can you give the `dispatchEvent` solution a shot? Normally in the end you should be able to get rid of the `useState` and `useEffect` in your implementation.

Reply:
> @wadjih-bencheikh18 can you give the `dispatchEvent` solution a shot?

Okay, I will try it

The reason why I used the `useEffect` and `useState` was the case where the `value` and `onChange` are not defined which I don't think `dispatchEvent` will solve it

Here's an example
https://github.com/zakodium-oss/react-science/blob/b66969a61a2ad6e946e6fbe264aeb2ed4907dcee/stories/components/input.stories.tsx#L85-L88

Reply:
ok i see

In that case I think we can have 2 controlled stories. One for controlled input and one for the uncontrolled input.

Reply:
I'm not sure if having an uncontrolled version of the input which re-renders on every value change because it has internal state tied to it really makes sense.

Maybe the `clearable` prop should only work for the controlled input and we log a warning when it's used inappropriately.

Reply:
But I'm also expecting it to work when it's uncontrolled. There are two steps in the solution I proposed:
- change the value programmatically
- create and send the event

Reply:
Using `dispatchEvent` did not update the state in case of controlled input
likely due to the issue discussed here: https://stackoverflow.com/a/39066491